File: Ruby/tests/HolidayValidator_test.rb
require 'minitest/autorun'
require_relative '../HolidayValidator'

class HolidayValidatorTest < Minitest::Test
  def setup
    @validator = HolidayValidator.new
  end

  def test_fixed_holidays
    assert @validator.holiday?(Date.new(2024, 4, 23))
  end

  def test_islamic_holidays
    assert @validator.holiday?(Date.new(2024, 4, 10))
  end
end

File: Rust/tests/holiday_validator_test.rs
use crate::holiday_validator::HolidayValidator;
use chrono::NaiveDate;

#[test]
fn test_fixed_holidays() {
    let validator = HolidayValidator::new(false);
    assert!(validator.is_holiday(NaiveDate::from_ymd(2024, 4, 23)));
}

#[test]
fn test_islamic_holidays() {
    let validator = HolidayValidator::new(false);
    assert!(validator.is_holiday(NaiveDate::from_ymd(2024, 4, 10)));
}
